docs/architecture/adr-069-gov-improvements.md

```diff
@@ -0,0 +1,228 @@
+(`spam` > `option_one + option_two + option_three + option_four` = proposal marked as spam).
+This allows clear spam proposals to be marked as spam easily, even with low participation from validators.
+
+To avoid voters wrongfully voting down a proposal as `SPAM`, voters will be slashed `x`% (default 0%) of their voting stake if they voted `SPAM` on a proposal that wasn't a spam proposal. The parameter allows to incentivise voters to only vote `SPAM` on actual spam proposals and not use `SPAM` as a way to vote `No with Veto` with a different threshold.
+
+This leads to the addition of the following governance parameter in [`v1.Params`][5]:
+
+```protobuf
+// burn_spam_amount defines the percentage of the voting stake that will be burned if a voter votes SPAM on a proposal that is not marked as SPAM.
+string burn_spam_amount = 8 [(cosmos_proto.scalar) = "cosmos.Dec"];
+```
+
+Additionally, the current vote options will be aliased to better accommodate the multiple choice proposal:
+
+```protobuf
+// VoteOption enumerates the valid vote options for a given governance proposal.
+enum VoteOption {
+  option allow_alias = true;
+
+  // VOTE_OPTION_UNSPECIFIED defines a no-op vote option.
+  VOTE_OPTION_UNSPECIFIED = 0;
+  // VOTE_OPTION_ONE defines the first proposal vote option.
+  VOTE_OPTION_ONE = 1;
+  // VOTE_OPTION_YES defines the yes proposal vote option.
+  VOTE_OPTION_YES = 1;
+  // VOTE_OPTION_TWO defines the second proposal vote option.
+  VOTE_OPTION_TWO = 2;
+  // VOTE_OPTION_ABSTAIN defines the abstain proposal vote option.
+  VOTE_OPTION_ABSTAIN = 2;
+  // VOTE_OPTION_THREE defines the third proposal vote option.
+  VOTE_OPTION_THREE = 3;
+  // VOTE_OPTION_NO defines the no proposal vote option.
+  VOTE_OPTION_NO = 3;
+  // VOTE_OPTION_FOUR defines the fourth proposal vote option.
+  VOTE_OPTION_FOUR = 4;
+  // VOTE_OPTION_NO_WITH_VETO defines the no with veto proposal vote option.
+  VOTE_OPTION_NO_WITH_VETO = 4;
+  // VOTE_OPTION_SPAM defines the spam proposal vote option.
+  VOTE_OPTION_SPAM = 5;
+}
+```
+
+The order does not change for a standard proposal (1 = yes, 2 = abstain, 3 = no, 4 = no with veto as it was) and the aliased enum can be used interchangeably.
+
+Updating vote options means updating [`v1.TallyResult`][5] as well.
+
+#### Tally
+
+Due to the vote option change, each proposal can have the same tallying method.
+
+However, chains may want to change the tallying function (weighted vote per voting power) of `x/gov` for a different algorithm (using a quadratic function on the voter stake, for instance).
+
+The custom tallying function can be passed to the `x/gov` keeper with the following interface:
+
+```go
+type Tally interface{
+    // to be decided
+
+    // Calculate calculates the tally result
+    Calculate(proposal v1.Proposal, govKeeper GovKeeper, stakingKeeper StakingKeeper) govv1.TallyResult
```

Comment:
I think Tally could be simplified to `Tally(ctx context.Context, govKeeper GovKeeper, prop v1.Proposal) (govv1.TallyResult, error)` 

I assume Tally could be a module's keeper implementing this interface and already holding within it other keepers besides gov. Passing GovKeeper in instead of having it in the tally implementer avoids circular dependencies between the tally interface and gov itself.

Reply:
The GovKeeper does not have the other keeps it uses public.
Which I think is good, we will be able to remove staking from the gov keeper as well altogether because it is only used for tallying.